main.go

```diff
@@ -29,11 +30,13 @@ type Module struct {
 
 type HTTPProbe struct {
 	// Defaults to 2xx.
-	ValidStatusCodes  []int  `yaml:"valid_status_codes"`
-	NoFollowRedirects bool   `yaml:"no_follow_redirects"`
-	FailIfSSL         bool   `yaml:"fail_if_ssl"`
-	FailIfNotSSL      bool   `yaml:"fail_if_not_ssl"`
-	Method            string `yaml:"method"`
+	ValidStatusCodes  []int    `yaml:"valid_status_codes"`
+	NoFollowRedirects bool     `yaml:"no_follow_redirects"`
+	FailIfSSL         bool     `yaml:"fail_if_ssl"`
+	FailIfNotSSL      bool     `yaml:"fail_if_not_ssl"`
+	Method            string   `yaml:"method"`
+	FailIfRegexp      []string `yaml:"fail_if_regexp"`
```

Comment:
As it contains more information, it is indeed easier to understand. I was shooting for consistency with the already-existing FailIfSSL. Do you want me to change it to FailIfMatchesRegexp?